src/ngraph/ngraph_imperative.cc

```diff
@@ -87,69 +87,128 @@ void NGImperative::parse_ngraph() {
+                         const std::vector<mxnet::OpReqType> &req,
+                         const std::vector<mxnet::NDArray> &outputs) -> void {
+            if (!compute_forward_imperative(attrs, ctx, inputs, req, outputs)) {
+              // use default mxnet compute kernel
+              fallbackx_fn(attrs, ctx, inputs, req, outputs);
+            }
+          },
+          11);
+    }
+    if (fallback_fn) {
+      op.set_attr<mxnet::FCompute>(
+          "FCompute<cpu>",
           [fallback_fn](const nnvm::NodeAttrs &attrs,
                         const mxnet::OpContext &ctx,
-                        const std::vector<mxnet::NDArray> &inputs,
+                        const std::vector<mxnet::TBlob> &inputs,
                         const std::vector<mxnet::OpReqType> &req,
-                        const std::vector<mxnet::NDArray> &outputs) -> void {
-            std::shared_ptr<Graph> op_ng;
-            int mode = static_cast<int>(GraphExeMode::kInfer);
-            if (!sparse_check(inputs) && !sparse_check(outputs)) {
-              // thread local cache for ngraph op
-              static thread_local NGIOpCache ngicache;
-              auto op_key = get_ngiop_key(attrs, ctx.run_ctx.ctx, inputs);
-              op_ng = ngicache[op_key];
-              if (!op_ng) {
-                NGImperative ngi(attrs, ctx.run_ctx.ctx, inputs, &req, outputs);
-                op_ng = ngicache[op_key] = ngi.get_op_ngraph();
-                if (ngraph_log_verbose && op_ng)
-                  LOG(INFO) << "Caching... " << attrs.op->name;
-              }
-              if (ctx.is_train) {
-                mode = static_cast<int>(GraphExeMode::kTrain);
-              }
-            }
-            if (op_ng && op_ng->ngraph_forward[mode]) {
-              compute_forward(ctx, op_ng, inputs, req, outputs);
-
-              if (ngraph_log_verbose_detail) {
-                LOG(INFO) << "ngraph imperative op: " << attrs.op->name
-                          << ", inputs " << std::to_string(inputs.size())
-                          << ", outputs " << std::to_string(outputs.size());
-
-                for (const auto &m : attrs.dict)
-                  LOG(INFO) << "attrs.dict[" << m.first << "] = " << m.second;
-              }
-            } else {
+                        const std::vector<mxnet::TBlob> &outputs) -> void {
+            std::vector<mxnet::NDArray> in;
+            for (auto &i : inputs) in.emplace_back(i, ctx.run_ctx.ctx.dev_id);
+            std::vector<mxnet::NDArray> out;
+            for (auto &i : outputs) out.emplace_back(i, ctx.run_ctx.ctx.dev_id);
+            if (!compute_forward_imperative(attrs, ctx, in, req, out)) {
               // use default mxnet compute kernel
               fallback_fn(attrs, ctx, inputs, req, outputs);
             }
           },
           11);
     }
+    if (ngraph_log_verbose_detail) {
+      if (!fallback_nd && !fallbackx_fn && !fallback_fn)
```

Comment:
I for one ~_welcome our new_~ would be grateful for there to be curly braces surrounding the `if` body.